[Benchmarks] Make detokenization optional in benchmark scripts

JArnoldAMD opened the pull request:
The benchmark scripts currently include detokenization as part of the performance measurement.  In most cases the detokenization time is minimal, but for some models and script parameters it can be significant.  Since detokenization isn't always needed (e.g. it may be handled by a separate server), it is helpful to have the option of whether or not to include detokenization in the measurement.

Since benchmark scripts for some other frameworks (notably TensorRT-LLM) do not include detokenization time, I have opted to skip detokenization by default.  This default could be changed if desired.

JArnoldAMD committed:
Make detokenization optional in benchmark scripts

Signed-off-by: Jeremy Arnold <Jeremy.Arnold@amd.com>

JArnoldAMD committed:
Set priority in the random inputs case

Signed-off-by: Jeremy Arnold <Jeremy.Arnold@amd.com>

JArnoldAMD committed:
Formatting updates

Signed-off-by: Jeremy Arnold <Jeremy.Arnold@amd.com>

JArnoldAMD committed:
Merge branch 'main' into benchmark-detokenization

JArnoldAMD committed:
Merge branch 'main' into benchmark-detokenization

tlrmchlsmth reviewed (commented):
This is a reasonable idea. I do think we should maintain the current behavior by default, which is to include detokenization in these benchmarks

JArnoldAMD commented:
> This is a reasonable idea. I do think we should maintain the current behavior by default, which is to include detokenization in these benchmarks

I'll update to change the default to do include detokenization, with a flag to disable it.

JArnoldAMD committed:
Switch to do detokenization by default, allow disable

Rather than disabling detokenization by default and allowing it to
be enabled, we now leave it enabled by default, and use a
`--disable-detokenize` flag to disable it.

Signed-off-by: Jeremy Arnold <Jeremy.Arnold@amd.com>

njhill reviewed (approved):
Thanks @JArnoldAMD, LGTM

njhill commented:
Now also supported in V1: https://github.com/vllm-project/vllm/pull/14224
